staging-next: 21.11 iteration 1

vcunat opened the pull request:
Another staging-next iteration. Documentation here: https://nixos.org/manual/nixpkgs/stable/#submitting-changes-staging-next-branch

https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents

*Per the [release schedule](https://discourse.nixos.org/t/lets-have-a-great-21-11-release-cycle/15262/3) we are still restricted to non-breaking changes to release critical packages, and the plan is to merge in about a week (Nov 12).*

Next staging-next: #146259

Artturin committed:
keybinder3: use gtk-x11 on darwin

trofi committed:
pytest: new hook to delete pytest bytecode

In https://github.com/NixOS/nixpkgs/issues/139292 we found out that
pytest leaves pytest-generated bytecode in install directory
(as tests are ran after install). That bytecode is:
- non-deterministic (was copied from cpython before cpython got
  support for deterministic bytecode)
- unneeded after test run

The change cleans bytecode up and provides a hook variable to avoid it
if needed.

Tested on `python39Packages.pytest-xdist` and `python27Packages.flaky` as:

    $ nix build -f. python39Packages.pytest-xdist
    $ nix build -f. python39Packages.pytest-xdist --rebuild

    $ nix build -f. python27Packages.flaky
    $ nix build -f. python27Packages.flaky --rebuild

Closes: https://github.com/NixOS/nixpkgs/issues/139292

felixsinger committed:
kexec-tools: 2.0.20 -> 2.0.22

Signed-off-by: Felix Singer <felixsinger@posteo.net>

Commit:
linuxPackages_5_10_hardened: fix patching conflict with patchShebang

sternenseemann committed:
fluidsynth_1: remove at 1.1.11

* FluidSynth 1.1.11 was kept around as a dependency of some packages
  that hadn't yet adjusted to API breakages. All of these packages now
  use FluidSynth 2.x, so fluidsynth_1 can be removed. It has been broken
  ever since glib was updated to 2.70 and was affected by an unpatched
  CVE.

* Refactor expression a bit, use pname instead of name.

* Add changelog entry in case someone was using this downstream
  (accidentally?).

Fixes #141508.
Fixes #124624.

sternenseemann committed:
python3Packages.zopfli: 0.1.8 -> 0.1.9

sternenseemann committed:
python3Packages.zopfli: link against our zopfli instead of vendoring

https://github.com/fonttools/py-zopfli/pull/9

legendofmiracles committed:
python3Packages.brotlipy: fix cross-compilation

legendofmiracles committed:
python3Packages.liquidctl: fix cross-compilation

legendofmiracles committed:
python3Packages.misaka: fix cross-compilation

legendofmiracles committed:
python3Packages.pycmarkgfm: fix cross-compilation

legendofmiracles committed:
python3Packages.pygit2: fix cross-compilation

legendofmiracles committed:
python3Packages.python-olm: fix cross-compilation

legendofmiracles committed:
python3Packages.reflink: fix cross-compilation

legendofmiracles committed:
python3Packages.smbus-cffi: fix cross-compilation

legendofmiracles committed:
python3Packages.soundfile: fix cross-compilation

legendofmiracles committed:
python3Packages.zstandard: fix cross-compilation

legendofmiracles committed:
python3Packages.editdistance-s: fix cross-compilation

legendofmiracles committed:
python3Packages.miniaudio: fix cross-compilation

legendofmiracles committed:
python3Packages.brotlicffi: fix cross-compilation

legendofmiracles committed:
python3Packages.xpybutil: fix cross-compilation

legendofmiracles committed:
python3Packages.xcffib: fix cross-compilation

legendofmiracles committed:
python3Packages.pykeepass: fix cross-compilation

legendofmiracles committed:
python3Packages.prox-tv: fix cross-compilation

legendofmiracles committed:
python3Packages.fastpbkdf2: fix cross-compilation

legendofmiracles committed:
python3Packages.cmarkgfm: fix cross-compilation

legendofmiracles committed:
python3Packages.cairocffi: fix cross-compilation, patch out unneeded tests

legendofmiracles committed:
python3Packages.cairosvg: fix cross-compilation, patch out unneeded tests

legendofmiracles committed:
python3Packages.tinycss2: 1.0.2 -> 1.1.0 + enable tests

legendofmiracles committed:
python3Packages.pynacl: add cffi dependency also at runtime

legendofmiracles committed:
python3Packages.argon2_cffi: fix cross-compilation, 20.1.0 -> 21.1.0

figsoda committed:
menyoki: 1.5.3 -> 1.5.5

legendofmiracles committed:
python3Packages.passlib: fix cross-compilation

arcz committed:
protobuf: 3.18.1 -> 3.19.0

Ma27 committed:
nixos/kexec: fix test

Ma27 committed:
nixos/tests: register `kexec' test

symphorien committed:
Merge pull request #142336 from yu-re-ka/feature-polkit-musl

polkit: fix musl build

trofi committed:
espeak-ng: explicitly disable parallel building

Current release lacks dependencies on local espeak-ng:

    cd dictsource && ESPEAK_DATA_PATH=/build/espeak-ng LD_LIBRARY_PATH=../src: \
        ../src/espeak-ng --compile=yue && cd ..
    bash: line 1: ../src/espeak-ng: No such file or directory

Should be fixed in next release: https://github.com/espeak-ng/espeak-ng/pull/1029
Can't apply patch as is as Makefile.am diverged quite a bit since 1.50
release.

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #142900 from trofi/sequential-espeak-ng

espeak-ng: explicitly disable parallel building

ivan committed:
ffmpeg_4: 4.4 -> 4.4.1

SuperSandro2000 committed:
Merge pull request #141918 from sternenseemann/remove-fluidsynth-1

trofi committed:
perlPackages.X11XCB: pull upstream fix for parallel build (#142642)

Co-authored-by: Zak B. Elep <zakame@zakame.net>

trofi committed:
perlPackages.libnet: fix Net/libnet.cfg determinism. (#142876)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #142777 from trofi/sequential-m17n_lib

github-actions[bot] committed:
Merge staging-next into staging

Artturin committed:
snappy: 1.1.8 -> 1.1.9 + add pkgconfig file

trofi committed:
diffoscope: adapt to file-5.41 failures

Pull in pustream fix for uImage diff test failure.

github-actions[bot] committed:
Merge staging-next into staging

figsoda committed:
rustPlatform: add support for features

figsoda committed:
ncspot: use buildFeatures

vcunat committed:
Merge #142684: gcc11: 11.1.0 -> 11.2.0 (into staging)

github-actions[bot] committed:
Merge staging-next into staging

marsam committed:
Merge pull request #142096 from arcz/protobuf-3.18.1

protobuf: 3.18.0 -> 3.19.0

7c6f434c committed:
Merge pull request #118100 from max-privatevoid/fix/glib-appinfo-watch

glib: follow profile updates when searching apps

figsoda committed:
lucky-commit: use checkNoDefaultFeatures

figsoda committed:
doc/rust: add documentation for feature options

marsam committed:
Merge pull request #142678 from marsam/update-libvpx

libvpx: 1.10.0 -> 1.11.0

legendofmiracles committed:
python3Packages.pycares: fix cross-compilation

SuperSandro2000 committed:
Merge pull request #142929 from ivan/ffmpeg-4.4.1

SuperSandro2000 committed:
Merge pull request #140475 from TredwellGit/xorg.libXi

SuperSandro2000 committed:
Merge pull request #141086 from TredwellGit/xorg.xf86inputlibinput

figsoda committed:
Merge pull request #143069 from figsoda/rust-features

[staging] rustPlatform: add support for features

Ma27 committed:
Merge pull request #141025 from felixsinger/pkgs/kexec-tools/update

kexec-tools: 2.0.20 -> 2.0.22

felixsinger committed:
graphviz: 2.47.3 -> 2.49.3 (#143231)

Signed-off-by: Felix Singer <felixsinger@posteo.net>

Ma27 committed:
Merge pull request #143221 from trofi/diffoscope-on-file-5.41

diffoscope: adapt to file-5.41 failures

alyssais committed:
Merge remote-tracking branch 'nixpkgs/staging-next' into staging

Conflicts:
	nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
	nixos/doc/manual/release-notes/rl-2111.section.md

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
gi-docgen: fix some docs build failures

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
glib: 2.70.0 → 2.70.1

jtojnar committed:
libsoup: 2.74.0 → 2.74.1

jtojnar committed:
libsoup_3: 3.0.1 → 3.0.2

jtojnar committed:
librsvg: 2.52.2 → 2.52.3

Mic92 committed:
python310Packages.ruamel-yaml-clib: 0.2.0 -> 0.2.4, add SuperSandro2000 as maintainer

github-actions[bot] committed:
Merge staging-next into staging

ttuegel committed:
qt515: 5.15.2 -> 5.15.3

Qt 5.15.3 does not have an official open source release, but the KDE team
maintains a collection of patches (pulled from Qt upstream) that they expect us
to carry.

figsoda committed:
synth: 0.6.0 -> 0.6.1

github-actions[bot] committed:
Merge staging-next into staging

figsoda committed:
epick: 0.5.1 -> 0.6.0

ttuegel committed:
qtbase: bootstrap with syncqt.pl

ttuegel committed:
qtModule: say the magic words

amaxine committed:
Merge pull request #143642 from jtojnar/more-gnome-staging

More GNOME updates (staging)

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
polkit: format the expression

Use lib.optionals instead of lib.optional, move phases to the bottom, have one input per line.

jtojnar committed:
polkit: 0.119 → 0.120

https://lists.freedesktop.org/archives/polkit-devel/2021-October/000608.html

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
python3Packages.httplib2: drop upstreamed patch (fails to apply)

```
$ nix build -f.  python3Packages.httplib2 -L
...
python3.9-httplib2> patching sources
python3.9-httplib2> applying patch /nix/store/gbpay7j261cbshpnkfgvp3lbnwa41p89-08d6993b69256fbc6c0b1c615c24910803c4d610.patch
python3.9-httplib2> patching file tests/test_http.py
python3.9-httplib2> Reversed (or previously applied) patch detected!  Assume -R? [n]
python3.9-httplib2> Apply anyway? [n]
python3.9-httplib2> Skipping patch.
python3.9-httplib2> 1 out of 1 hunk ignored -- saving rejects to file tests/test_http.py.rej
```

jtojnar committed:
polkit: port to Meson

Autotools build will be removed in the next release.

Examples are no longer installed.

Musl patch does not work.

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

r-burns committed:
libselinux: fix cross-compilation from x86_32

libselinux incorrectly initializes the host platform's arch using uname -m,
which actually returns the build platform's arch. This results in an invalid
compile flag (-mno-tls-direct-seg-refs) when compiling from x86_32.
So we need to set the host arch variable explicitly.

github-actions[bot] committed:
Merge staging-next into staging

marsam committed:
icu70: init at 70.1

marsam committed:
icu: update homepage

marsam committed:
icu: 69.1 -> 70.1

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
Merge pull request #143834 from jtojnar/polkit

polkit: 0.119 → 0.120

github-actions[bot] committed:
Merge staging-next into staging

ttuegel committed:
sddm: fix build with Qt 5.15.3

ttuegel committed:
qt515: combine source locks into single JSON file

ttuegel committed:
qt515: consolidate extra package sources

ttuegel committed:
qt515: remove unused file

ttuegel committed:
qt515: consolidate bash set short options

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

ttuegel committed:
qtModule: combine conditionals inside evaluation brackets

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

ttuegel committed:
qtbase: avoid rebuilding versions < 5.15

github-actions[bot] committed:
Merge staging-next into staging

7c6f434c committed:
Merge pull request #144005 from marsam/update-icu

icu: 69.1 -> 70.1

SuperSandro2000 committed:
Merge pull request #143809 from trofi/fix-httplib2

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
mtr: fix build against upcoming ncurses-6.3

On ncurses-6.3 with extra printf() annotations gcc now detects
use of user input in place of format strings:

    ui/curses.c:765:42:
      error: format not a string literal and no format arguments [-Werror=format-security]
      765 |         mvprintw(rowstat - 1, startstat, msg);
          |                                          ^~~

7c6f434c committed:
Merge pull request #144080 from trofi/fix-mtr-for-ncurses-6.3

mtr: fix build against upcoming ncurses-6.3

Artturin committed:
Merge pull request #143058 from Artturin/snapppypkgconfig

SuperSandro2000 committed:
python39Packages.ruamel_yaml: remove python2 compatility from python3 file, add SuperSandro2000 as maintainer

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #120760 from sternenseemann/ncurses-no-paches

Artturin committed:
gtk3: restore x11 support on darwin

jtojnar committed:
polkit: Fix regression in paths in pkg-config file

resulting from discrepancy between Meson and Autotools build systems.

jtojnar committed:
polkit: fix hash

alyssais committed:
rust: 1.56.0 -> 1.56.1

I've tested building fd on x86_64-linux.  Because this release is so
tiny[1], I'm confident skipping the usual, bigger, tests we do for
Rust.

[1]: https://github.com/rust-lang/rust/compare/1.56.0...1.56.1

SuperSandro2000 committed:
Merge pull request #143640 from Mic92/ruamel-yaml-clib

github-actions[bot] committed:
Merge staging-next into staging

primeos committed:
clang-analyzer: Fix the build with LLVM 13

We changed the source structure ("src") by switching from individual
source tarballs to the mono repository. Therefore we have to prepend
"clang/" now.

SuperSandro2000 committed:
python39Packages.ruamel-base: normalise name, adopt

SuperSandro2000 committed:
python39Packages.ruamel-ordereddict: normalise name, adopt

SuperSandro2000 committed:
python39Packages.ruamel-yaml: normalise name, adopt

SuperSandro2000 committed:
python39Packages.ruamel-yaml-clib: normalise name, adopt

primeos committed:
Merge pull request #140783 from primeos/llvmPackages_latest

llvmPackages_latest: 12 -> 13

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #144204 from alyssais/rust-1.56.1

vcunat committed:
gnutls: avoid the (check-time) cacert dependency

The point is to reduce rebuild amount when updating cacert/nss,
though at *this* point it remains quite high - before & after:

Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
  13109 x86_64-darwin
  21567 x86_64-linux
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
  13109 x86_64-darwin
  17141 x86_64-linux

ttuegel committed:
Merge pull request #143852 from ttuegel/qt-5.15.3

Qt 5.15.3

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #144160 from SuperSandro2000/ruamel

figsoda committed:
Merge pull request #143688 from figsoda/update-synth

synth: 0.6.0 -> 0.6.1

figsoda committed:
Merge pull request #142625 from figsoda/update-menyoki

menyoki: 1.5.3 -> 1.5.5

figsoda committed:
Merge pull request #143713 from figsoda/update-epick

epick: 0.5.1 -> 0.6.0

r-burns committed:
Merge pull request #126557 from r-burns/autogen

[staging] autogen: fix patchelf invocation to remove generic build path from rpath

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
Merge remote-tracking branch 'origin/staging-next' into staging

github-actions[bot] committed:
Merge master into staging-next

r-ryantm committed:
elan: 1.3.0 -> 1.3.1

figsoda committed:
cargo-about: 0.3.0 -> 0.4.1

github-actions[bot] committed:
Merge master into staging-next

alyssais committed:
Merge remote-tracking branch 'nixpkgs/staging-next' into staging

SuperSandro2000 committed:
Merge pull request #138618 from trofi/update-ell

ell: 0.43 -> 0.44

gebner committed:
Merge pull request #144327 from r-ryantm/auto-update/elan

elan: 1.3.0 -> 1.3.1

SuperSandro2000 committed:
treewide: remove python39Packages.ruamel_yaml aliases

Artturin committed:
Merge pull request #132239 from Artturin/keybinder3-darwin

github-actions[bot] committed:
Merge master into staging-next

figsoda committed:
Merge pull request #144333 from figsoda/update-cargo-about

cargo-about: 0.3.0 -> 0.4.1

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #144407 from SuperSandro2000/ruamel_yaml

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

Artturin committed:
Merge pull request #143914 from r-burns/libselinux

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

r-burns committed:
python3Packages.click: 8.0.2 -> 8.0.3

Also fix unmatched substitute call, which has been failing since the
code was formatted with black for v7.1.

alyssais committed:
pkgsMusl.dbus: enable systemd support

We previously weren't able to build systemd for Musl, but now we
can!  (But not statically.)  So there's no longer any reason to have
systemd support in D-Bus disabled by default for pkgsMusl.

alyssais committed:
pkgsMusl.redis: enable systemd support

We previously weren't able to build systemd for Musl, but now we
can!  (But not statically.)  So there's no longer any reason to have
systemd support in Redis disabled by default for pkgsMusl.

alyssais committed:
pkgsMusl.cups: enable systemd support

We previously weren't able to build systemd for Musl, but now we
can!  (But not statically.)  So there's no longer any reason to have
systemd support in CUPS disabled by default for pkgsMusl.

jian-lin committed:
linux: enable TCP_CONG_ADVANCED

TCP_CONG_ADVANCED is enabled by default on x86_64[1] in the upstream.
Although it is not the case for aarch64[2], many distributions, such as
Debian[3], Fedora[4] and Gentoo[5], choose to enable it in their
distribution kernel.

With this patch, aarch64 users can choose many other TCP congestion
algorithms, which may improve their network performance.

[1]: https://github.com/torvalds/linux/blob/7ddb58cb0ecae8e8b6181d736a87667cc9ab8389/arch/x86/configs/x86_64_defconfig#L71
[2]: https://github.com/torvalds/linux/blob/7ddb58cb0ecae8e8b6181d736a87667cc9ab8389/arch/arm64/configs/defconfig
[3]: https://salsa.debian.org/kernel-team/linux/-/blob/e2d14375d74cdcb2cfed6518d7a5ad784c30df1a/debian/config/config#L7063
[4]: https://src.fedoraproject.org/rpms/kernel/raw/836165dd2dff34e4f2c47ca8f9c803002c1e6530/f/kernel-aarch64-fedora.config
[5]: https://github.com/gentoo/gentoo/blob/5808eb2f066ea2ece19c35a1978b7f79077a467b/sys-kernel/gentoo-kernel/gentoo-kernel-5.10.77.ebuild#L27

vcunat committed:
Merge #144223: gnutls: avoid cacert dependency

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #144585 from r-burns/click

SuperSandro2000 committed:
Merge pull request #142221 from sternenseemann/zopfli-0.1.9

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

lovesegfault committed:
Merge pull request #144641 from jian-lin/enable-TCP_CONG_ADVANCED-on-aarch64

linux: enable TCP_CONG_ADVANCED

vcunat committed:
Merge branch 'staging-next' into staging

bobby285271 commented on pkgs/applications/misc/khard/default.nix:
```diff
@@ -18,7 +18,7 @@ python3.pkgs.buildPythonApplication rec {
     vobject
-    ruamel_yaml
+    ruamel-yaml
     ruamel_base
```
https://github.com/NixOS/nixpkgs/pull/144407#issuecomment-959834865

```suggestion
    ruamel-base
```

SuperSandro2000 replied:
Fixing that right now. This can be entirely removed because ruamel-yaml propgogtes it.

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

SuperSandro2000 committed:
khard: fix eval, add import check

figsoda committed:
kmon: 1.5.5 -> 1.6.0, clarify license

SuperSandro2000 committed:
Merge pull request #144786 from SuperSandro2000/khard

figsoda committed:
Merge pull request #144792 from figsoda/update-kmon

kmon: 1.5.5 -> 1.6.0, clarify license

github-actions[bot] committed:
Merge master into staging-next

fabianhjr committed:
kexec-tools: 2.0.22 → 2.0.23

github-actions[bot] committed:
Merge master into staging-next

Izorkin commented:
Curl not updating to version 7.79.1?

github-actions[bot] committed:
Merge master into staging-next

happysalada commented:
> Curl not updating to version 7.79.1?

If you look at the diff, the update is included. However, the github interface for this PR does not show the commit. I don't understand why.

mweinelt commented:
106e944e19d6fdbb0fc3d64495403b788835e1d4

![image](https://user-images.githubusercontent.com/131599/140610669-0fa5072f-1750-48e7-ae14-0bc585aac20d.png)

vcunat commented:
The GitHub's UI just isn't meant to handle PRs with hundreds of commits 🤷🏽

github-actions[bot] committed:
Merge master into staging-next

figsoda committed:
synth: fix darwin build

figsoda committed:
Merge pull request #144925 from figsoda/fix-synth

synth: fix darwin build

r-burns committed:
snappy: fix build on x86_64-darwin

Needs patch for compatibility with clang 7

github-actions[bot] committed:
Merge master into staging-next

happysalada commented:
Just switched my darwin system to this branch and couldn't find any regressions so far. Of course it doesn't prove anything, but it's a good sign I think.

fabianhjr commented:
Ditto for linux-x86_64 since yesterday but required one version bump to build and boot: #144823.

github-actions[bot] committed:
Merge master into staging-next

trofi committed:
diffoscope: drop upstreamed fix-uimage-on-file-5.41.patch patch

diffoscope-190 already contains the fix:

    applying patch /nix/store/icfygfs9d249x7if0w7004vimzkj4082-fix-uimage-on-file-5.41.patch
    patching file tests/comparators/test_uimage.py
    Hunk #1 FAILED at 98.
    1 out of 1 hunk FAILED -- saving rejects to file tests/comparators/test_uimage.py.re

github-actions[bot] committed:
Merge master into staging-next

dasJ committed:
Merge pull request #144976 from trofi/fix-diffoscope-patch

diffoscope: drop upstreamed fix-uimage-on-file-5.41.patch patch

github-actions[bot] committed:
Merge master into staging-next

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into staging-next

SuperSandro2000 committed:
Merge pull request #144941 from r-burns/snappy

r-burns committed:
Merge pull request #144823 from LibreCybernetics/update-kexec-tools-staging-next

[staging-next] kexec-tools: 2.0.22 → 2.0.23

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

alyssais committed:
zellij: 0.19.0 -> 0.20.0

figsoda committed:
Merge pull request #145200 from alyssais/zellij

[staging-next] zellij: 0.19.0 -> 0.20.0

happysalada commented:
Thank you for updating zellij! I thought we had to wait until this was merged in master before doing small updates. If that is alright, there is another rust package that needs rust 1.56 to compile. If it's not okay, feel free to let me know, I'll wait.

github-actions[bot] committed:
Merge master into staging-next

fabianhjr commented:
Hi @happysalada, I think it is ok to do so. Check the discussion around this topic on #137501

happysalada committed:
helix: 0.4.1 -> 0.5.0

SuperSandro2000 committed:
Merge pull request #145232 from happysalada/update_helix

happysalada committed:
starship: 0.58.0 -> 1.0.0

marsam committed:
Merge pull request #145263 from happysalada/update_starship

starship: 0.58.0 -> 1.0.0

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

happysalada committed:
zellij: 0.20.0 -> 0.20.1

github-actions[bot] committed:
Merge master into staging-next

figsoda committed:
Merge pull request #145388 from happysalada/update_zellij

zellij: 0.20.0 -> 0.20.1

jonringer reviewed (approved):
From the linux side, this has 0 x86_64-linux build regressions:

```
$ nix-env --option system x86_64-linux -f /home/jon/.cache/nixpkgs-review/pr-144730-1/nixpkgs -qaP --xml --out-path --show-trace --meta
10 packages added:
icu69 (init at 69.1) lldb_12 (init at 12.0.1) lvm2-2_02 (init at 2.02.187) protobuf3_18 (init at 3.18.1) python38Packages.ruamel_base (init at 1.0.0) python38Packages.ruamel_yaml (init at 0.17.16) python38Packages.ruamel_yaml_clib (init at 0.2.4) python39Packages.ruamel_base (init at 1.0.0) python39Packages.ruamel_yaml (init at 0.17.16) python39Packages.ruamel_yaml_clib (init at 0.2.4)

35607 packages updated:
...
```

oxalica commented:
I'm on 25b9e19da39ab9fd81452d1d5d0042acedd24567, `firefox-unwrapped` failed to build
Log: https://gist.github.com/oxalica/e9e563b0ee44b68b86185536f93c4a1d

fabianhjr commented:
Hi @oxalica, it has been reported on and there is some ongoing discussion on: https://github.com/NixOS/nixpkgs/issues/144670

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

oxalica committed:
libreoffice-still: 7.1.6.2 -> 7.1.7.2

This contains the fix for build error with bision 3.8.
See https://gerrit.libreoffice.org/c/core/+/122082

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

oxalica committed:
firefox: add a patch fixing deadlock

github-actions[bot] committed:
Merge master into staging-next

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into staging-next

Conflicts were caused by 2043dbb6faa9e21b0fb500161542e30d6c8bc680
interacting with b2767b69ddc702eb926633a5f2336534352cb7b8.

github-actions[bot] committed:
Merge master into staging-next

oxalica committed:
firefox: use rustc.llvmPackages.stdenv with bintools

Currently `clangUseLLVM` is broken since it uses libcxx and compiler-rt but
also specifies `--gcc-toolchain`, which leads to weird compile errors when
including C++ headers.

oxalica committed:
libreoffice*: skip test to fix build with icu70

nrdxp commented:
Looks like we are a day behind on this. Anything holding up the merge? I'd like to help get this merged by EOD at least, if possible.

github-actions[bot] committed:
Merge master into staging-next

fabianhjr commented:
@nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.

Jon Ringer reported no regressions otherwise.

fabianhjr commented:
Regarding that issue, it seems like there is some approval for a fix: https://github.com/NixOS/nixpkgs/pull/145459 (3 hours ago) that should ideally be merged into staging-next.

sternenseemann commented:
> @nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.

There are other big ones like libreoffice and thunderbird, likely more. As it stands we absolutely *can't* merge as the `unstable` jobset doesn't succeed.

Seems like Jon's build report doesn't really hold anymore.

vcunat commented:
Hydra's take on this, though some new failures were apparently temporary and will disappear on restart: https://hydra.nixos.org/eval/1721869?compare=1721771

I noticed the libreoffice regression mentioned a bit above, too.  Nothing else stood out to me, though in the comparison there still are over 10k unfinished builds.

jonringer commented:
> > @nrdxp, not sure if a blocker but the only issue I am aware of is firefox: #144670.
> 
> There are other big ones like libreoffice and thunderbird, likely more. As it stands we absolutely _can't_ merge as the `unstable` jobset doesn't succeed.
> 
> Seems like Jon's build report doesn't really hold anymore.

It was my third time attempting to review it. if there are flakey  builds then it's possible i just got lucky a few times.

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

Mic92 committed:
Merge pull request #145589 from oxalica/fix/libreoffice

[staging-next] libreoffice*: update and fix build

vcunat committed:
Revert "metrics: drop requiredSystemFeatures; /cc #76776"

This reverts commit c7aef6dedb29392ef2aad8fe70fa9745597c84ea.

I believe it should work again since today, though I'm being a bit
conservative, pushing to `staging-next` instead of `master`.

github-actions[bot] committed:
Merge master into staging-next

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into staging-next

sternenseemann commented:
@ofborg eval

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into staging-next

sternenseemann commented:
Merged again to include evaluation fix bdaf941e39921dc7ef0da9c8263c45943c850eec for the issue introduced in 6fe99723d3c23d64942e4de893ce5053198686bd while resolving a conflict in the first merge.

github-actions[bot] committed:
Merge master into staging-next

yu-re-ka commented:
Runs on my systems just fine now. Firefox and Thunderbird work as well.

`hyperscan`, which is optionally used by `rspamd`, doesn't build.

vcunat commented:
https://hydra.nixos.org/eval/1722344?compare=1722301#tabs-now-fail

We're certainly not perfect, but to me this PR seems OK for `master`.
